pkg/controllers/pod/pod_controller.go

```diff
@@ -0,0 +1,310 @@
+package pod
+
+import (
+	log "github.com/Sirupsen/logrus"
+	calicocache "github.com/projectcalico/k8s-policy/pkg/cache"
+	"github.com/projectcalico/k8s-policy/pkg/controllers/controller"
+	"github.com/projectcalico/k8s-policy/pkg/converter"
+	"github.com/projectcalico/libcalico-go/lib/api"
+	"github.com/projectcalico/libcalico-go/lib/client"
+	"github.com/projectcalico/libcalico-go/lib/errors"
+	"k8s.io/apimachinery/pkg/fields"
+	uruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/pkg/api/v1"
+	"k8s.io/client-go/tools/cache"
+	"reflect"
+)
+
+// PodController Implements Controller interface
+// Responsible for monitoring kubernetes pods and
+// updating pod labels to calico ETCD if changed.
+type PodController struct {
+	indexer       cache.Indexer
+	informer      cache.Controller
+	labelCache    calicocache.ResourceCache
+	endpointCache map[string]api.WorkloadEndpoint
+	calicoClient  *client.Client
+	k8sClientset  *kubernetes.Clientset
+}
+
+// NewPodController Constructor for PodController
+func NewPodController(k8sClientset *kubernetes.Clientset, calicoClient *client.Client) controller.Controller {
+
+	podConverter := converter.NewPodConverter()
+
+	// Function returns map of podName:podObject stored by policy controller
```

Comment:
This comment is wrong too - it doesn't return podName -> podObject.

It returns a map of key -> WorkloadEndpoint